File: src/imovel.js
const Status = {
  DISPONIVEL: "disponível",
  CAPTADO: "captado",
  VISITADO: "visitado",
  NEGOCIANDO: "negociando",
  FECHADO: "fechado",
};

const TipoImovel = {
  CASA: "casa",
  APARTAMENTO: "apartamento",
  TERRENO: "terreno",
  COMERCIAL: "comercial",
};

class Imovel {
  constructor({ endereco, tipo, preco, areaM2, quartos = 0, status = Status.DISPONIVEL, observacoes = "" }) {
    this.endereco = endereco;
    this.tipo = tipo;
    this.preco = preco;
    this.areaM2 = areaM2;
    this.quartos = quartos;
    this.status = status;
    this.observacoes = observacoes;
    this.dataCadastro = new Date();
  }

  atualizarStatus(novoStatus) {
    this.status = novoStatus;
  }

  resumo() {
    const tipo = this.tipo.charAt(0).toUpperCase() + this.tipo.slice(1);
    const preco = this.preco.toLocaleString("pt-BR", { style: "currency", currency: "BRL" });
    const data = this.dataCadastro.toLocaleDateString("pt-BR");
    return `${tipo} - ${this.endereco}\nPreço: ${preco} | Área: ${this.areaM2}m² | Quartos: ${this.quartos}\nStatus: ${this.status} | Cadastro: ${data}`;
  }
}

module.exports = { Imovel, Status, TipoImovel };

File: src/imovel.py
from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum


class Status(Enum):
    DISPONIVEL = "disponível"
    CAPTADO = "captado"
    VISITADO = "visitado"
    NEGOCIANDO = "negociando"
    FECHADO = "fechado"


class TipoImovel(Enum):
    CASA = "casa"
    APARTAMENTO = "apartamento"
    TERRENO = "terreno"
    COMERCIAL = "comercial"


@dataclass
class Imovel:
    endereco: str
    tipo: TipoImovel
    preco: float
    area_m2: float
    quartos: int = 0
    status: Status = Status.DISPONIVEL
    observacoes: str = ""
    data_cadastro: datetime = field(default_factory=datetime.now)

    def atualizar_status(self, novo_status: Status):
        self.status = novo_status

    def resumo(self) -> str:
        return (
            f"{self.tipo.value.title()} - {self.endereco}\n"
            f"Preço: R$ {self.preco:,.2f} | Área: {self.area_m2}m² | Quartos: {self.quartos}\n"
            f"Status: {self.status.value} | Cadastro: {self.data_cadastro:%d/%m/%Y}"
        )
